check/controls.go

```diff
@@ -40,60 +47,93 @@ type Summary struct {
 	Info int `json:"total_info"`
 }
 
+func (controls *Controls) WithIsAction(isAction bool) *Controls {
```

Comment:
See below my comment about functional chaining... it's OK, but it's a heavyweight way to assign some fields!